### ISSUE TITLE
docker-compose example nginx.conf produces card/caldav error

### ISSUE THREAD

**Issue by kowjens** [nextcloud/docker#927]
The actual example nginx.conf under

[.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/fpm/web/nginx.conf](https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/fpm/web/nginx.conf)

contains a change in the card/calldav redirecting section (line 81 and 85):

    return 301 $scheme://$host:$server_port/remote.php/dav;

If I use this I get an error in the _settings->overviev_ page inside nextcloud saying I should consult the docs:

[docs.nextcloud.com #service-discovery](https://docs.nextcloud.com/server/17/admin_manual/issues/general_troubleshooting.html#service-discovery)

I checked my old config and found the `:$server_port` part is causing the problem. This fixes the problem:

    return 301 $scheme://$host/remote.php/dav;

for both `location = /.well-known/carddav` and accordingly in the calldav section.

Someone else found this issue or is this wanted and just caused by my setup :) ?

kr Jens

**Comment by dbroeh** [comment]
Same here. Reverting to the old state fixed the issue also for me.

**Comment by am97** [comment]
Same here, removing `:$server_port` also solved the problem for me.

**Comment by monreal** [comment]
Not sure if this is the same but after updating to nc18 now I see the following in setting and security warnings:

> 
>     Your web server is not properly set up to resolve "/.well-known/caldav". Further information can be found in the documentation.
>     Your web server is not properly set up to resolve "/.well-known/carddav". Further information can be found in the documentation.

**Comment by p-baum** [comment]
I can confirm monreals issue using docker hub image.

**Comment by leggettc18** [comment]
Can confirm the same is true for me with the nextcloud alpine-fpm image with nginx reverse-proxy

**Comment by e-ru** [comment]
same here

a `curl -I https://example.com/.well-known/carddav`

responds with

```
...
location: http://example.com:80/remote.php/dav
...
```
when changing `location = /.well-known/carddav ...` and `location = /.well-known/caldav` to 

`return 301 https://$host/remote.php/dav;`

then the warning disappears.

My question here is why are the nginx variables `$scheme` and `$server_port` are resolved to `http` and `80`. Does it belongs to the fact that i'm using a proxy?

My setup is basically this one:

[.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/fpm](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/fpm)

**Comment by Tockra** [comment]
Okay, I have the same issue. 
Replacing $scheme by "https" and removing $server_port saluted my problem.
But I think its ugly. Can I just change`server_port` and `scheme` to the needed values?

**Comment by am97** [comment]
> Replacing $scheme by "https" and removing $server_port saluted my problem.
> But I think its ugly. Can I just change`server_port` and `scheme` to the needed values?

Just removing `$server_port` should be enough. You should have: 
```
        location = /.well-known/carddav {
            return 301 $scheme://$host/remote.php/dav;
        }

        location = /.well-known/caldav {
            return 301 $scheme://$host/remote.php/dav;
        }
```

By the way, this issue has been fixed here: https://github.com/nextcloud/docker/pull/1380

**Comment by Tockra** [comment]
What I'm doing wrong, I don't see that patch at:
https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy-self-signed-ssl/mariadb/fpm/web/nginx.conf

**Comment by joshtrichards** [comment]
Fixed in #1409
